src/mesh.cpp

```diff
@@ -235,6 +259,28 @@ bool loadPlyFile(const QString& fileName,
     }
     if (info.colors.size() != info.verts.size())
         info.colors.clear();
+
+    // Reconstruct inner polygons
+    while (innerPolygonVertexCount.size())
+    {
+        unsigned int verticesCount = innerPolygonVertexCount.back();
+        unsigned long firstIdx = info.edges.size();
+        for (size_t j = 0; j < verticesCount; ++j)
+        {
+            if (j > 1)
+                info.edges.push_back(info.edges.back());
+            info.edges.push_back(innerVertexIndices.back());
+            innerVertexIndices.pop_back();
```

Comment:
Again, I think it would be more natural to iterate over `innerVertexIndices` (in forward order) rather than erasing elements from it.  Especially since the way it's written here the order of everything gets reversed by starting from the back.  It's not wrong, but I did find it confusing.